pymc_marketing/clv/models/beta_geo_beta_binom.py

```diff
@@ -0,0 +1,670 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+from collections.abc import Sequence
+from typing import Literal
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+import xarray
+from numpy import exp
+from pymc.util import RandomState
+from scipy.special import betaln, gammaln
+
+from pymc_marketing.clv.distributions import BetaGeoBetaBinom
+from pymc_marketing.clv.models.basic import CLVModel
+from pymc_marketing.clv.utils import to_xarray
+from pymc_marketing.model_config import ModelConfig
+from pymc_marketing.prior import Prior
+
+
+# TODO: Docstring Examples
+class BetaGeoBetaBinomModel(CLVModel):
+    """Beta-Geometric/Beta-Binomial Model (BG/BB) for non-contractual, discrete purchase opportunities,
+    introduced by Fadel et al. [1]_.
+
+    The BG/BB model assumes the dropout process across the customer population follows a Beta distribution,
+    and time between purchases is also Beta-distributed while customers are still active.
```

Comment:
I think this is wrong? Dropout process follows a geometric distribution and probability of purchase while alive is bernoulli distributed, so time between transactions would actually be geometric as well?